shimmingtoolbox/cli/check_env.py

```diff
@@ -191,7 +191,7 @@ def get_sct_version() -> str:
     if sct_version.returncode != 0:
         raise subprocess.CalledProcessError("Error while getting SCT's version")
     version_output: str = sct_version.stdout.rstrip()
-    version_output = version_output.split("\n\n")[2].split("\n")[1]
+    version_output = version_output.split("\n\n")[2].split("\n")[-5]
```

Comment:
Oh my goodness, I had no idea that ADS was fetching SCT's version string by parsing the plaintext output of `sct_check_dependencies`.

I would recommend using the `sct_version` command instead, which outputs the information you're looking for on its own:

```console
joshua@tadpole:~$ sct_version
git-mgp/remove-coverage-b3d00edffa43d49a7bac0f9b3ddcc8d570334edd
```

Reply:
I haven't been a dev here, I'm just a new user, so I didn't know either! I think this won't work out of the box though, because their test seems to be looking for the `-version` flag, 

https://github.com/shimming-toolbox/shimming-toolbox/blob/4013cae6031e544e5a3ae77e2b623e17a14b90af/test/cli/test_cli_check_env.py#L82-L88

but that could be fixed too if the actual devs feel it's a better solution. I don't know if their regex for `- version` is used anywhere else in the code.

Reply:
I agree that using `sct_version` would be way cleaner, especially as we might expect the parsed string to change again in the future (and therefore the [-5] index might not be suited anymore). Is there a chance that `sct_version` could not be available with early versions of sct ? Even if I doubt that people will have very old versions installed this could still be a potential concern...

Reply:
Ah, the git history there is a bit misleading. In december 2020, the entire `scripts` folder was moved. GitHub does a poor job at displaying history when the file location changes, so it ends the history there.

`sct_version` was introduced May 2020 (https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/2723) while `sct_check_dependencies` has been present since the very first SCT commit (https://github.com/spinalcordtoolbox/spinalcordtoolbox/commit/10f9a5d5628669c7a8cdbe565a5b57b50c72b488).

Reply:
Two suggestions:

* Perhaps you could `try:` using `sct_version`, and if the command isn't present, fall back to checking `sct_check_dependencies`?
* For `sct_check_dependencies`, perhaps you could filter for lines that start with `- version:`, that way it might would work regardless of the position in the output?

Reply:
From what I understand, [-5] should work with present and past versions because the version string has always been located 5 lines from the end of the output. However if in the future some new line is appended to the end of the output, it won't work anymore

Reply:
Ah, my apologies. Please disregard my comment! You're absolutely right.

Reply:
So, as I understand it, the consensus is that we should just try resolving this bug using `sct_version`? As I mentioned, both the `sct_version` and `sct_check_depencencies` command were created the same day (even though the behaviours of `sct_check_depencencies` existed elsewhere in SCT in prior versions), so at the very least the changes made in this PR will be retrocompatible up to the earliest version of `sct_check_depencencies`, plus all versions since the recent bug (on our end) was introduced.

I'll make the changes ASAP and push it so that the PR can go ahead and the ST tests stop failing.

Reply:
I've made the change and changed the title. The only issue now is that the old was that the project was testing this was to look for a line that included the string "version", but our new output doesn't contain that string anymore (this was included in the output of sct_check_depencencies but not sct_version). 

See lines:

https://github.com/shimming-toolbox/shimming-toolbox/blob/4013cae6031e544e5a3ae77e2b623e17a14b90af/test/cli/test_cli_check_env.py#L82-L88

I *think* the proper version output should always start with `git` now, so maybe we could test for that instead? Here is the output for me locally:

```
(Pdb) print(version_output)
git-mb/3810-mtsat-7d495ca33892ccb20148ec5e6f1b9650c1f3f2fe
```

I'll try that and see if the CI tests work on GitHub with it.

Reply:
Done in [e8d330c](https://github.com/shimming-toolbox/shimming-toolbox/pull/402/commits/e8d330cb562aba48ddef28ea450651b061d8b73c)

Reply:
> I _think_ the proper version output should always start with `git` now, so maybe we could test for that instead? Here is the output for me locally:

One thing to maybe keep in mind is that `sct_version` is written so that if a stable release version of SCT is installed, it will output just the release string (e.g. `5.6`). 

However, since the version string check is only relevant in the context of ST's test suite, I think this shouldn't ever be an issue, given that ST's test suite only ever clones SCT:

https://github.com/shimming-toolbox/shimming-toolbox/blob/4013cae6031e544e5a3ae77e2b623e17a14b90af/.github/workflows/tests.yml#L55-L58

And I think we can safely assume that if the test suite can fetch `git-` version strings, then it can fetch any kind of version string (`git-`, `5.6`, etc.) in production. So, this LGTM!

Reply:
>So, this LGTM!

Great, thanks! Can you approve so this can be merged?